tests/test_bar.py

```diff
@@ -1,67 +1,164 @@
+from functools import partial
+from typing import Tuple
+
 import numpy as np
 import pymbar
 import pytest
-from pymbar.testsystems import ExponentialTestCase, gaussian_work_example
+from numpy.typing import NDArray
+from pymbar.testsystems import ExponentialTestCase
 
 from timemachine.fe.bar import (
     bar_with_bootstrapped_uncertainty,
     bootstrap_bar,
     compute_fwd_and_reverse_df_over_time,
+    df_and_err_from_u_kln,
+    df_from_u_kln,
     pair_overlap_from_ukln,
+    works_from_ukln,
 )
 
 
+def make_gaussian_ukln_example(
+    params_a: Tuple[float, float], params_b: Tuple[float, float], seed: int = 0, n_samples: int = 2000
+) -> Tuple[NDArray, float]:
+    """Generate 2-state u_kln matrix for a pair of normal distributions."""
+
+    def u(mu, sigma, x):
+        return (x - mu) ** 2 / (2 * sigma ** 2)
+
+    mu_a, sigma_a = params_a
+    mu_b, sigma_b = params_b
+
+    u_a = partial(u, mu_a, sigma_a)
+    u_b = partial(u, mu_b, sigma_b)
+
+    rng = np.random.default_rng(seed)
+
+    x_a = rng.normal(mu_a, sigma_a, (n_samples,))
+    x_b = rng.normal(mu_b, sigma_b, (n_samples,))
+
+    u_kln = np.array([[u_a(x_a), u_a(x_b)], [u_b(x_a), u_b(x_b)]])
+
+    dlogZ = np.log(sigma_a) - np.log(sigma_b)
+
+    return u_kln, dlogZ
+
+
+@pytest.fixture
+def partial_overlap_uniform_ukln_example():
+    def u_a(x):
+        """Unif[0.0, 1.0], with log_Z = 0"""
+        in_bounds = (x > 0) * (x < 1)
+        return np.where(in_bounds, 0, +np.inf)
+
+    def u_b(x):
+        """Unif[0.5, 1.5], with log_Z = -5"""
+        x_ = x - 0.5
+        return u_a(x_) + 5.0
+
+    rng = np.random.default_rng(2023)
+
+    x_a = rng.uniform(0, 1, (1000,))
+    x_b = rng.uniform(0.5, 1.5, (1000,))
+
+    assert np.isfinite(u_a(x_a)).all()
+    assert np.isfinite(u_b(x_b)).all()
+
+    u_kln = np.array([[u_a(x_a), u_a(x_b)], [u_b(x_a), u_b(x_b)]])
```

Comment:
unsure of ordering of first two dimensions, don't remember convention

Reply:
Added a note on convention in https://github.com/proteneer/timemachine/pull/1098/commits/0ed9719e5d2fad979457649857a2e2ec16d32d67

Reply:
>based on the convention used in timemachine (k indexes evaluation state, l indexes sampling state); however, it seems like pymbar might use the opposite convention

Ahh, I think usages in timemachine may be mixed: another usage I'm aware of is consistent with the one in pymbar.utils.kln_to_kn https://github.com/proteneer/timemachine/blob/c4369bdd3accbb88d48cabd24c9dc209d19d94f6/timemachine/maps/estimators.py#L54-L61

Doesn't matter which convention is used if used uniformly (just a transpose), but mixed conventions could lead to bugs. (May be improved in the future by picking less ambiguous named indices, can afford more than a single letter per dimension.)

Reply:
Created an issue to track this: https://github.com/proteneer/timemachine/issues/1100